calculator.rb

```diff
@@ -0,0 +1,79 @@
+# add numbers 
+def add(first_num, second_num)
+  return (first_num) + (second_num)
+end 
+
+# subtract numbers
+def subtract(first_num, second_num)
+  return (first_num) - (second_num)
+end 
+
+# multiply numbers 
+def multiply(first_num, second_num)
+  return (first_num) * (second_num)
+end 
+
+# divide numbers 
+def divide(first_num, second_num)
+  if second_num == 0
+    puts "You can't divide by zero."
+    exit 
+  else
+    return (first_num / second_num)
+  end
+end 
+
+# checking if the string from the beginning to the end contains only digits one or more times 
+def check_valid_number(user_number)
+  # if user input contains a digit, convert to an integer. otherwise, display error message
+  if /^\d+$/.match(user_number)
```

Comment:
Good use of a custom method to check for valid user input. Consider how you might change your regular expression to allow for negative numbers.